### PR TITLE
add forward-first option for forward zones.

### PR DESCRIPTION
**Pull request by RyanFolsom**
Small change to allow for the forward-first option to be applied to forward zones.

### CODE REVIEW

File: manifests/forward.pp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 define unbound::forward (
   $address,
   $zone = $name,
+  $forward_first = 'no'
```
Comment (xaque208) [inline manifests/forward.pp]: I'd like to see a validation around this for accepted values of 'yes' and 'no'.  Also please align `=`.

File: templates/forward.erb [hunk 2]
```diff
@@ -5,5 +5,6 @@ forward-zone:
   forward-addr: <%= addr %>
 <% end -%>
 <% elsif @address != '' -%>
-  forward-addr: <%= @address -%>
+  forward-addr: <%= @address %>
 <% end -%>
+  forward-first: <%= @forward_first -%>
```
Comment (xaque208) [inline templates/forward.erb]: Could we only include this line if the value of `@forward_first` is not the default?  This is really just for cleaner configs.

File: manifests/forward.pp (hunk 1 above)
Comment (RyanFolsom) [inline manifests/forward.pp]: Can do. Is input validate being done elsewhere through the stdlib functions? Will I need to update the requirements section under README.md?

Comment (xaque208) [inline manifests/forward.pp]: Yes and yes.  The `validate_re()` function can help.

Comment (RyanFolsom) [inline manifests/forward.pp]: Want me to validate_array on $address while I'm in there? Looks like the template will only use it if it's an array.

Comment (xaque208) [inline manifests/forward.pp]: That also sounds like a good idea.

Comment (RyanFolsom) [inline manifests/forward.pp]: I was wrong about that - the template does work if it's not an array.

```puppet
<% if @address.is_a? Array -%>
<% @address.each do |addr| -%>
  forward-addr: <%= addr %>
<% end -%>
<% elsif @address != '' -%>
  forward-addr: <%= @address %>
<% end -%>
```
Sorry about that!

File: templates/forward.erb [hunk 3]
```diff
@@ -5,5 +5,8 @@ forward-zone:
   forward-addr: <%= addr %>
 <% end -%>
 <% elsif @address != '' -%>
-  forward-addr: <%= @address -%>
+  forward-addr: <%= @address %>
+<% end -%>
+<% if @forward_first != 'no' -%>
+  forward-first: <%= @forward_first -%>
```
Comment (xaque208) [inline templates/forward.erb]: The trailing - here in the erb may not be desired.

**Comment by xaque208** [comment]
This looks reasonable.  If I'm wrong about the trailing -, this gets a +1 from me.

**Comment by igalic** [comment]
you're right, @xaque208. that looks like my mistake. we don't need to propagate it.

**Comment by RyanFolsom** [comment]
Following up - are there any other modifications you would like to see made to this PR?

**Comment by xaque208** [comment]
Apologies for the delay, I'm just returning from holiday.  This looks good, thank you.

**Comment by RyanFolsom** [comment]
No worries! Hope you enjoyed your holiday. Thank you for the merge!

**Pull request merged**